src/librustc/session/config.rs

```diff
@@ -949,6 +950,7 @@ pub fn default_configuration(sess: &Session) -> ast::CrateConfig {
         mk(InternedString::new("target_pointer_width"), intern(wordsz)),
         mk(InternedString::new("target_env"), intern(env)),
         mk(InternedString::new("target_vendor"), intern(vendor)),
+        mk(InternedString::new("llvm_target"), intern(llvm_target)),
```

Comment:
Note that this is *not* necessary. We can do conditional compilation for Cortex-M targets just fine without this change **iff** the Cortex-M are named `thumbv*` because we have to disambiguate betwen Cortex-M and our others ARM+Linux targets. This means we can no longer name them e.g. `armv7m-none-eabi`, which may be less strange to users, for who "thumb" may not ring a bell.